### PR TITLE
feat(preset-wind): support linear-gradient percent

### PR DESCRIPTION
**Pull request by Simon-He95**
close: #2423 

image

### CODE REVIEW

File: packages/preset-wind/src/rules/background.ts [hunk 1]
```diff
@@ -22,6 +22,11 @@ const bgGradientColorValue = (mode: string, cssColor: CSSColorValue | undefined,
 
 const bgGradientColorResolver = (mode: 'from' | 'to' | 'via') =>
   ([, body]: string[], { theme }: RuleContext<Theme>) => {
+    if (/^[0-9]/.test(body)) {
+      return {
+        [`--un-gradient-${mode}-position`]: h.bracket(`[${body}]`)
```
Comment (chu121su12) [inline packages/preset-wind/src/rules/background.ts:27]: You may want to do var reset since using this without stops now may generate unset variable.

Comment (Simon-He95) [reply packages/preset-wind/src/rules/background.ts:27]: I have a problem the default value will cover the set value. How to resolve it?

image

Comment (Simon-He95) [reply packages/preset-wind/src/rules/background.ts]: I fix it through add another rule

File: packages/preset-wind/src/rules/background.ts [hunk 2]
```diff
@@ -77,7 +86,9 @@ export const backgroundStyles: Rule[] = [
   [/^(?:bg-gradient-)?from-op(?:acity)?-?(.+)$/, ([, opacity]) => ({ '--un-from-opacity': h.bracket.percent(opacity) })],
   [/^(?:bg-gradient-)?via-op(?:acity)?-?(.+)$/, ([, opacity]) => ({ '--un-via-opacity': h.bracket.percent(opacity) })],
   [/^(?:bg-gradient-)?to-op(?:acity)?-?(.+)$/, ([, opacity]) => ({ '--un-to-opacity': h.bracket.percent(opacity) })],
-
+  [/from-([0-9].+)$/, bgGradientPositionResolver('from')],
```
Comment (antfu) [inline packages/preset-wind/src/rules/background.ts:89]:
```suggestion
  [/^(from|via|to)-([\d\.]+)%$/, bgGradientPositionResolver('from')],
```

Like this? Should we explicitly ask for it to be end with `%`?

File: packages/preset-wind/src/rules/background.ts [hunk 3]
```diff
@@ -30,29 +30,38 @@ const bgGradientColorResolver = (mode: 'from' | 'to' | 'via') =>
     const { alpha, color, cssColor } = data
 
     if (!color)
-      return
+      return {}
```
Comment (antfu) [inline packages/preset-wind/src/rules/background.ts:33]: Why do we return this?

File: packages/preset-wind/src/rules/background.ts (hunk 2 above)
Comment (Simon-He95) [reply packages/preset-wind/src/rules/background.ts]: Of course, I have adjusted

**Pull request merged**